pkg/template/jsonpath.go

```diff
@@ -0,0 +1,155 @@
+package template
+
+import (
+	"bytes"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"reflect"
+	"regexp"
+
+	"k8s.io/client-go/third_party/forked/golang/template"
+	"k8s.io/client-go/util/jsonpath"
+)
+
+var (
+	tektonVar  = regexp.MustCompile(`^\$\((.*)\)$`)
+	jsonRegexp = regexp.MustCompile(`^\{\.?([^{}]+)\}$|^\.?([^{}]+)$`)
+)
+
+func ParseJSONPath(input interface{}, expr string) (string, error) {
+	j := jsonpath.New("").AllowMissingKeys(false)
+	buf := new(bytes.Buffer)
+
+	//First turn the expression into fully valid JSONPath
+	expr, err := TektonJSONPathExpression(expr)
+	if err != nil {
+		return "", err
+	}
+
+	if err := j.Parse(expr); err != nil {
+		return "", err
+	}
+
+	fullResults, err := j.FindResults(input)
+	if err != nil { //
+		return "", err
+	}
+
+	// Copy of j.PrintResults
+	for ix := range fullResults {
+		if err := printResults(buf, fullResults[ix]); err != nil {
```

Comment:
Can you provide guidance on how to interpret fullResults (`[][]reflect.Value`) vs results (`[]reflect.Value`)? Is this a distinction of line vs field?

Reply:
so I think the only time the length of `[][]reflect.Value` is greater than 1 is when we use `range` keyword. AFAIK the `range` keyword is not standard JSONPath, its something that is only available in this library and used in kubectl

Reply:
https://gist.github.com/so0k/42313dbb3b547a0f51a547bb968696ba#jsonpath